src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AbstractEnchantmentMachine.java

```diff
@@ -55,4 +61,16 @@ protected void showEnchantmentLevelWarning(@Nonnull BlockMenu menu) {
         menu.replaceExistingItem(22, progressBar);
     }
 
+    protected boolean hasIgnoredLore(@Nonnull ItemStack itemStack) {
+        List<String> ignoredLore = ignoredLores.getValue();
+        if (itemStack.hasItemMeta()) {
+            List<String> itemLore = itemStack.getItemMeta().getLore();
```

Comment:
getLore can be null so your contains check can throw an NPE. You will want to do a != null check here or do hasLore (which does != null && !isEmpty)

Reply:
hasLore() preferabbly.

Reply:
Once we wandered between whether to do hasLore. Fixed now